### ISSUE TITLE
`expiring-todo-comments` should include the comment itself in the report 

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#401]
Copied from https://github.com/eslint/eslint/issues/12327

---

I think the point of this rule is to keep track of TODO comments, however this is hard to follow; I'd have to open each file separately to see if I can handle a task.

# Current

```ts
  source/globals.d.ts:1:1
  ⚠    1:1   Unexpected todo comment.                                                                   no-warning-comments
  ⚠    2:1   Unexpected todo comment.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   Unexpected todo comment.                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   Unexpected todo comment.                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   Unexpected todo comment.                                                                   no-warning-comments
```

# Ideal

```ts
  source/globals.d.ts:1:1
  ⚠    1:1   Unexpected todo comment: Drop ignore after #9.                                                                  no-warning-comments
  ⚠    2:1   Unexpected todo comment: Find this info on the page.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   Unexpected todo comment: Improve JSX types for event listeners so we can use... [clipped]                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   Unexpected todo comment: Drop some definitions when their related bugs are resolved                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   Unexpected todo comment: Drop `as` after https://github.com/microsoft/TSJS-lib-generator/issues/741                                                                   no-warning-comments
```

Or

```ts
  source/globals.d.ts:1:1
  ⚠    1:1   TODO: Drop ignore after #9.                                                                  no-warning-comments
  ⚠    2:1   FIXME: Find this info on the page.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   TODO: Improve JSX types for event listeners so we can use... [clipped]                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   TODO: Drop some definitions when their related bugs are resolved                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   TODO: Drop `as` after https://github.com/microsoft/TSJS-lib-generator/issues/741                                                                   no-warning-comments
```

**Comment by sindresorhus** [comment]
// @lubien In case you're interested in working on this.

**Comment by sindresorhus** [comment]
I would go for the short version:

> ⚠    1:1   TODO: Drop ignore after #9.

**Comment by lubien** [comment]
> // @lubien In case you're interested in working on this.

I can work on this weekend, sure!



> I would go for the short version:

Noted.

How long until clip? I could use something like `process.stdout.columns` to calculate the width and try not to overflow too much. I'd have also to count ESLint own input.

```sh
lubien@heron:~/dev$ npx eslint Untitled-1.js

~/dev/foo.js
  1:1  error  Unexpected 'todo' comment                                    unicorn/expiring-todo-comments

✖ 1 problem (1 errors, 0 warnings)

lubien@heron:~/dev$ node
> process.stdout.columns
103
```

Given 103 column digits. For this case I can see 4 digits for reporting level, a few for `line:column` (in this case just 3, but would need more). Double space separators at first then some spaces to make the rule name appear at right. 61 digits for the report message.

How about making messages at most 50% `process.stdout.columns`? With a fallback width in case some situation make this value not available (happened to me at least in Elixir I don't know about NodeJS).

**Comment by sindresorhus** [comment]
We can’t depend on process.stdout.columns as we can’t know how the reporter will format it. We just have to decide on an arbitrary (large) limit and truncate.

**Comment by sindresorhus** [comment]
> I can work on this weekend, sure!

Are you still interested in working on this? No pressure at all. Just want to open this issue up for other if you're busy :)

**Comment by lubien** [comment]
@sindresorhus Interested for sure but I'm afraid my free time wont be enough for a couple weeks or so. 

Sorry for delaying this but would be best if we open this issue for another person for now but I'm eager to return contributing soon anytime by December so whenever I can help please tag me.

**Comment by fregante** [comment]
FYI, this was implemented and released in eslint: https://github.com/eslint/eslint/issues/12327

**Comment by fisker** [comment]
I'm updating dependencies in #816
